src/strategy/saml_client.ts

```diff
@@ -70,9 +77,17 @@ export class CustomSamlClient extends PassportSaml.SAML {
       .map(tamperAuthorizeRequest => (e: Error, xml?: string) => {
         xml
           ? tamperAuthorizeRequest(xml)
-              .chain(tamperedXml =>
-                this.extededCacheProvider.save(tamperedXml, this.config)
-              )
+              .chain(tamperedXml => {
+                if (this.logCallback !== undefined) {
+                  this.logCallback(
+                    requestIp.getClientIp(req),
+                    tamperedXml,
+                    new Date().toISOString(),
+                    true
+                  );
+                }
```

Comment:
this is not a good place to put the callback since this code runs only if tamperAuthorizeRequest is defined

Reply:
Do you mean I should use `xml` before it is tampered?